client/src/main/java/org/apache/celeborn/client/ShuffleClientImpl.java

```diff
@@ -126,6 +126,7 @@ public class ShuffleClientImpl extends ShuffleClient {
   private final boolean authEnabled;
   private final TransportConf dataTransportConf;
 
+  @SuppressWarnings("ThreadLocalUsage")
   private final ThreadLocal<Compressor> compressorThreadLocal =
```

Comment:
@mridulm, because `initialValue` depends on `conf`, `compressorThreadLocal` could not use `static`. Therefore this just use `@SuppressWarnings("ThreadLocalUsage")`.